src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImpl.java

```diff
@@ -5,55 +5,159 @@
 import com.ait.lienzo.client.core.event.NodeDragStartEvent;
 import com.ait.lienzo.client.core.event.NodeMouseClickEvent;
 import com.ait.lienzo.client.core.event.NodeMouseDoubleClickEvent;
+import com.ait.lienzo.client.core.shape.wires.SelectionManager;
 import com.ait.lienzo.client.core.shape.wires.WiresConnector;
 import com.ait.lienzo.client.core.shape.wires.WiresManager;
 import com.ait.lienzo.client.core.shape.wires.handlers.WiresConnectorControl;
 import com.ait.lienzo.client.core.shape.wires.handlers.WiresConnectorHandler;
+import com.ait.tooling.common.api.java.util.function.Consumer;
+import com.google.gwt.user.client.Timer;
 
 public class WiresConnectorHandlerImpl implements WiresConnectorHandler {
 
-    private final WiresConnectorControl m_control;
-
     private final WiresConnector m_connector;
-
     private final WiresManager m_wiresManager;
+    private final Consumer<NodeMouseClickEvent> clickEventConsumer;
+    private final Consumer<NodeMouseDoubleClickEvent> doubleClickEventConsumer;
+
+    public WiresConnectorHandlerImpl(final WiresConnector connector,
+                                     final WiresManager wiresManager) {
+        this(connector,
+             wiresManager,
+             new DefaultClickHandlers(wiresManager, connector));
+    }
 
-    public WiresConnectorHandlerImpl(WiresConnector connector, WiresManager wiresManager) {
-        this.m_control = wiresManager.getControlFactory().newConnectorControl(connector, wiresManager);
+    public WiresConnectorHandlerImpl(final WiresConnector connector,
+                                     final WiresManager wiresManager,
+                                     final DefaultClickHandlers defaultClickHandlers) {
+        this(connector,
+             wiresManager,
+             defaultClickHandlers.onClick(),
+             defaultClickHandlers.onDoubleClick());
+    }
+
+    public WiresConnectorHandlerImpl(final WiresConnector connector,
+                                     final WiresManager wiresManager,
+                                     final Consumer<NodeMouseClickEvent> clickEventConsumer,
```

Comment:
The Counsumers could be of class that represents the data used by the action that will be taken, not the Event itself.